Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,267 @@
+        /// <param name="message">The request message.</param>
+        /// <returns>The user input.</returns>
+        public static string PromptString(string message)
+        {
+            IsValidName(PromptInput(message), out string result);
+            return result;
+        }
+
+        /// <summary>
+        /// Calculates the BMI by weight[kg] and height[m].
+        /// </summary>
+        /// <param name="weightInKg">Weight in kilograms.</param>
+        /// <param name="heightInM">Height in meters.</param>
+        /// <returns>Returns the BMI or -1 if at least one of the parameters was invalid.</returns>
+        public static float CalculateBmi(float weightInKg, float heightInM)
+        {
+            string errorMessage = "Failed calculating BMI. Reason:";
+
+            bool isInvalidWeight = IsLessEqualZero(weightInKg, "Weight", ref errorMessage);
+            bool isInvalidHeight = IsLessEqualZero(heightInM, "Height", ref errorMessage, !isInvalidWeight);
+            
+            if (isInvalidWeight || isInvalidHeight)
+            {
+                Console.WriteLine(errorMessage);
+                return InvalidNumberInput;
+            }
+
+            return weightInKg / heightInM / heightInM;
+        }
+
+        /// <summary>
+        /// Reads in the data of a person and processes it further.
+        /// </summary>
+        private static void ProcessPerson()
+        {
+            string fullName = PromptString("What is your full name? ");
+            int age = PromptInt("What is your age? ");
+            float weightInKg = PromptFloat("What is your weight in kg? ");
+            float heightInCm = PromptFloat("What is your height in cm? ");
+
+            Console.WriteLine($"{fullName} is {age} years old, his weight is {weightInKg} kg and his height is {heightInCm} cm.");
+            Console.WriteLine($"His BMI is {CalculateBmi(weightInKg, heightInCm / 100)}.");
+        }
+
+        /// <summary>
+        /// Prints the message and reads in the user input.
+        /// </summary>
+        /// <param name="message">The message to print.</param>
+        /// <returns>The unparsed and unvalidated input.</returns>
+        private static string PromptInput(string message)
+        {
+            Console.WriteLine(message);
+            return Console.ReadLine();
+        }
+
+        /// <summary>
+        /// Prompts for an integer value. Converts the string input to an integer if possible.
+        /// </summary>
+        /// <param name="message">The message to print before the prompt.</param>
+        /// <param name="result">The parsing result. Is 0 if the parsing failed.</param>
+        /// <param name="input">The user input.</param>
+        /// <returns>Returns true if the parsing succeeded, false otherwise.</returns>
+        private static bool PromptTryParse(string message, out int result, out string input)
+        {
+            return int.TryParse(input = PromptInput(message), out result);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Extract the assignment of 'input' from this expression.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885761)